platform/mbed_assert.c

```diff
@@ -13,15 +13,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+#include <string.h>
 #include "platform/mbed_assert.h"
-#include "device.h"
 
-#include "platform/mbed_interface.h"
 #include "platform/mbed_critical.h"
+#include "platform/mbed_error.h"
 
 void mbed_assert_internal(const char *expr, const char *file, int line)
 {
     core_util_critical_section_enter();
-    mbed_error_printf("mbed assertation failed: %s, file: %s, line %d \n", expr, file, line);
-    mbed_die();
```

Comment:
Why does this function no longer need mbed_die()?

@deepikabhavnani @SenRamakri

Reply:
Nvm, looked over the PR again and it looks ok.

Reply:
mbed_die is called but after error processing, just like any other error, next addition to removal is 
mbed_error()
https://github.com/ARMmbed/mbed-os/blob/e6ae3d2c5168158e3ab879913f79317d416cd5fa/platform/mbed_error.c#L200